src/main/java/seedu/duke/command/RecordCommand.java

```diff
@@ -23,18 +25,48 @@ public RecordCommand(Ui ui, Data data, HashMap<String, String> arguments) {
     @Override
     public void execute() throws Exception {
         assert ui != null : "Ui must not be null";
-        assert arguments.containsKey("payload") : "Arguments must contain a value for the `payload` key";
+        assert arguments.containsKey(Constants.PAYLOAD_KEY) : "Arguments must contain a value for the `payload` key";
         // TODO: Implement proper exception
         Patient patient = data.currentPatient;
         if (patient == null) {
             throw new Exception(Constants.EXCEPTION_RECORD_RETRIEVE_NULLPATIENT);
         }
-        if (arguments.get("payload").length() == 0) {
-            throw new Exception(Constants.EXCEPTION_RECORD_EMPTYPAYLOAD);
+        String dateString = arguments.get(Constants.PAYLOAD_KEY);
+        LocalDate date = null;
+        try {
+            date = parseDate(dateString);
+        } catch (DateTimeParseException dateTimeParseException) {
+            throw new Exception(Constants.EXCEPTION_RECORD_RETRIEVE_INVALID_DATE);
+        }
+        addRecord(patient, date);
+        printNewRecord(patient);
+    }
+
+    private LocalDate parseDate(String dateString) throws DateTimeParseException {
+        if (!dateString.isEmpty()) {
+            return LocalDate.parse(dateString);
```

Comment:
Probably you would like to use Singapore time format instead of the default ISO one for both printing and parsing.
```java
protected static String datePattern = "dd/MM/yyyy";
...
LocalDate date = LocalDate.parse(dateTimeString, DateTimeFormatter.ofPattern(datePattern));
System.out.println(date.format(DateTimeFormatter.ofPattern(datePattern));
```

Reply:
Sounds good, will use this format.